cmd/register/files.go

```diff
@@ -48,50 +62,28 @@ Usage
 )
 
 func registerFromFilesFunc(ctx context.Context, args []string, cmdCtx cmdCore.CommandContext) error {
-	files := args
-	sort.Strings(files)
-	logger.Infof(ctx, "Parsing files... Total(%v)", len(files))
-	logger.Infof(ctx, "Params version %v", filesConfig.version)
-	var registerResults [] RegisterResult
-	adminPrinter := printer.Printer{}
+	dataRefs, tmpDir, err := getSortedFileList(ctx, args)
+	if err != nil {
+		logger.Errorf(ctx, "error while un-archiving files in tmp dir due to %v", err)
+		return nil
+	}
+	logger.Infof(ctx, "Parsing files... Total(%v)", len(dataRefs))
 	fastFail := !filesConfig.skipOnError
-	logger.Infof(ctx, "Fail fast %v", fastFail)
 	var _err error
-	for i := 0; i< len(files) && !(fastFail && _err != nil) ; i++ {
-		absFilePath := files[i]
-		var registerResult RegisterResult
-		logger.Infof(ctx, "Parsing  %v", absFilePath)
-		fileContents, err := ioutil.ReadFile(absFilePath)
-		if err != nil {
-			registerResult =  RegisterResult{Name: absFilePath, Status: "Failed", Info: fmt.Sprintf("Error reading file due to %v", err)}
-			registerResults = append(registerResults, registerResult)
-			_err = err
-			continue
-		}
-		spec, err := unMarshalContents(ctx, fileContents, absFilePath)
+	var registerResults [] RegisterResult
+	for i := 0; i < len(dataRefs) && !(fastFail && _err != nil) ; i++ {
+		fileContents, err := ioutil.ReadFile(dataRefs[i])
 		if err != nil {
-			registerResult =  RegisterResult{Name: absFilePath, Status: "Failed", Info: fmt.Sprintf("Error unmarshalling file due to %v", err)}
-			registerResults = append(registerResults, registerResult)
-			_err = err
-			continue
-		}
-		if err := hydrateSpec(spec); err != nil {
-			registerResult =  RegisterResult{Name: absFilePath, Status: "Failed", Info: fmt.Sprintf("Error hydrating spec due to %v", err)}
-			registerResults = append(registerResults, registerResult)
 			_err = err
 			continue
 		}
-		logger.Debugf(ctx, "Hydrated spec : %v", getJsonSpec(spec))
-		if err := register(ctx, spec, cmdCtx); err != nil {
-			registerResult =  RegisterResult{Name: absFilePath, Status: "Failed", Info: fmt.Sprintf("Error registering file due to %v", err)}
-			registerResults = append(registerResults, registerResult)
-			_err = err
-			continue
-		}
-		registerResult =  RegisterResult{Name: absFilePath, Status: "Success", Info: "Successfully registered file"}
-		registerResults = append(registerResults, registerResult)
+		registerResults, _err = registerContent(ctx, fileContents, dataRefs[i], registerResults, cmdCtx)
 	}
 	payload, _ := json.Marshal(registerResults)
-	adminPrinter.JSONToTable(payload, projectColumns)
+	registerPrinter := printer.Printer{}
+	registerPrinter.JSONToTable(payload, projectColumns)
+	if _err = os.RemoveAll(tmpDir); _err != nil {
```

Comment:
is tmpdir always set and if so should we always delete it. It should only be deleted in the archive/download case right?